Improve match patterns and add host permissions section to permissions (#6943)
GoogleChrome/developer.chrome.com: site/en/docs/extensions/mv3/declare_permissions/index.md

---

To use most [Chrome APIs][api-ref], your extension must declare its intent in the permissions fields

> of the [manifest](#manifest). Extensions can request the following categories of permissions,specified using the respective manifest keys:

Reviewer: ```suggestion
of the [manifest](#manifest). Extensions can request the following categories of permissions, specified using the respective manifest keys:
```

[`"permissions"`](#permissions)

> : contain items from a list of known strings (such as `"geolocation"`). Changes may trigger a [warning](#warnings).

Reviewer 2: ```suggestion
: contains items from a list of known strings (such as `"geolocation"`). Changes may trigger a [warning](#warnings).
```

Reviewer 2: Please do this throughout.

[`"optional_permissions"`][api-perms]
: are like regular `permissions`, but are granted by the user at runtime, rather than in advance.

[`"content_scripts.matches"`][doc-cs-static]
: contain one or more [match patterns][doc-match] that allows content scripts to inject into one or more hosts. Changes may trigger a [warning](#warnings).

[`"host_permissions"`](#host-permissions)
: contain one or more [match patterns][doc-match] that give access to one or more hosts. Changes may trigger a [warning](#warnings).

`"optional_host_permissions"`

> : are like regular `host_permissions`, but are granted by the user at runtime, rather than in advance.

Reviewer: ```suggestion
: are like regular `host_permissions`, but are granted by the user at runtime, rather than at install time.
```

Not sure about this

}
```

## Host permissions {: #host-permissions }

> Host permissions allow extensions to interact with the URLs [matching patterns][doc-match]. Some [Chrome APIs][api-ref] require host permissions in addition to their own API permission. Host permissions allow extensions to:

Reviewer 2: ```suggestion
Host permissions allow extensions to interact with the URL's [matching patterns][doc-match]. Some [Chrome APIs][api-ref] require host permissions in addition to their own API permission. Host permissions allow extensions to:
```

Reviewer 2: >Host permissions allow extensions to:

Can you future proof this a bit? This is the kind of thing we'll forget to update when new host permissions are added. Specifically, I think you should replace this with a statement that goes something like "Host permissions are documented with the APIs that require them. A few examples are described below."

> - Make [`fetch()`][mdn-fetch] requests from the service worker and extension pages.

Reviewer 2: ```suggestion
- Make [`fetch()`][mdn-fetch] requests from the extension service worker and extension pages.
```
